core/src/Cabana_Tuple.hpp

```diff
@@ -144,6 +144,30 @@ template <typename... Types>
 struct Tuple<MemberTypes<Types...>> : SoA<MemberTypes<Types...>, 1>
 {
     using base = SoA<MemberTypes<Types...>, 1>;
+
+    KOKKOS_FORCEINLINE_FUNCTION Tuple() = default;
+
+    KOKKOS_FORCEINLINE_FUNCTION Tuple( const Tuple &t )
+    {
+        Impl::tupleCopy( *this, 0, t, 0 );
+    }
+
+    KOKKOS_FORCEINLINE_FUNCTION Tuple( const Tuple &&t )
```

Comment:
```suggestion
    KOKKOS_FORCEINLINE_FUNCTION Tuple( Tuple &&t )
```

Reply:
Good catch---this also needs to go into `operator=( Tuple &&t )`. Our original implementation also zeroed `t`, which we determined wasn't necessary because `std::move` only requires the source variable to be in a well-defined state (more relevant for cases where an object does resource management).

Adding `const` probably came along for the ride out of habit once we got rid of the explicit zeroing.

Reply:
Sorry guys - bad habits...